orc8r/cloud/easy_deploy/docker/docker_root/scripts/deploy_cli.py

```diff
@@ -0,0 +1,108 @@
+            constants = yaml.load(f, Loader=yaml.FullLoader)
+    except OSError:
+        print("Failed opening config.yml file")
+
+def run_subprocess(cmd : List[str]):
+    proc = subprocess.Popen(cmd,
+                            stdin=subprocess.PIPE,
+                            stdout=subprocess.PIPE,
+                            )
+    try:
+        output, err = proc.communicate()
+        if output:
+            pprint.pprint(output.decode().split('\n'))
+
+        if err:
+            pprint.pprint(err.decode().split('\n'))
+    except subprocess.SubprocessError as err:
+        proc.kill()
+        sys.exit("Error: {}".format(err))
+
+def parse_config(args):
+    if not os.path.isdir(constants["configs_dir"]):
+        try:
+            os.makedirs(constants["configs_dir"])
+        except OSError as error:
+            print("failed creating config directories ", error)
+            sys.exit(1)
+    configure_all(constants, args)
+
+def parse_install(args):
+    playbook = "%s/playbooks/main.yml" % constants["scripts_dir"]
+    if args.precheck:
+        tags = "precheck"
+    elif args.addcerts:
+        tags = "addcerts"
+    cmd = ["ansible-playbook", "-v", "-e", "@/root/config.yml", "-t", tags, playbook]
+    run_subprocess(cmd)
+
+def parse_upgrade(args):
+    pass
+
+def parse_cleanup(args):
+    pass
+
+def parse_verify(args):
+    pass
+
+def add_config_parser(subparsers):
+    config_parser = subparsers.add_parser('configure')
+    config_parser.set_defaults(func=parse_config)
+    config_type_group = config_parser.add_mutually_exclusive_group()
+    for config_type in (INFRA_CONFIG_TYPE,
+                        PLATFORM_CONFIG_TYPE,
+                        SERVICE_CONFIG_TYPE):
+        config_type_group.add_argument('--%s' % config_type, action='store_true')
+
+    action_type_group = config_parser.add_mutually_exclusive_group()
+    action_type_group.add_argument('--check', action='store_true', help="check if mandatory configs are set")
+    action_type_group.add_argument('--show', action='store_true', help="show existing configuration")
+    action_type_group.add_argument('--info', action='store_true', help="show all configuration options")
+
+def add_install_parser(subparsers):
+    install_subparser = subparsers.add_parser('install')
+    install_subparser.set_defaults(func=parse_install)
+    install_type_group = install_subparser.add_mutually_exclusive_group()
+    for config_type in (INFRA_CONFIG_TYPE,
+                        PLATFORM_CONFIG_TYPE,
+                        SERVICE_CONFIG_TYPE):
+        install_type_group.add_argument('--%s' % config_type, action='store_true')
+    action_type_group = install_subparser.add_mutually_exclusive_group()
+    action_type_group.add_argument('--precheck', action='store_true')
+    action_type_group.add_argument('--addcerts', action='store_true')
+
+if __name__ == "__main__":
+    load_constants()
+    parser = argparse.ArgumentParser()
```

Comment:
If we're intending this CLI to grow over time, perhaps it makes sense to use a more declarative approach to defining it, e.g. https://click.palletsprojects.com/en/7.x/

Reply:
Good point. I have heard good things about https://github.com/google/python-fire as well. Will work on either of these.